adding marqtune model names to model_registry.py

adityabharadwaj198 opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Currently there are models defined in marqo repository but they are not defined using the hugging face standard (Marqtune defines using hugging face standard). This change fixes that 

* **What is the current behavior?** (You can also link to an open issue here)
Marqtune model names are defined in Marqo's own standard

* **What is the new behavior (if this is a feature change)?**
* Marqtune model names will be defined with hugging face standard also


* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
* NA


* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
* Yes


* **Related Python client changes** (link commit/PR here)
* NA


* **Related documentation changes** (link commit/PR here)
* NA


* **Other information**:
* NA


* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

adityabharadwaj198 committed:
adding marqtune model names to model_registry.py

adityabharadwaj198 committed:
adding marqtune model names to tests

adityabharadwaj198 committed:
Updating embeddings_open_clip_python_3_8.json file

farshidz commented on tests/backwards_compatibility_tests/test_general_compatibility.py:
```diff
@@ -8,7 +8,7 @@
 class GeneralCompatibilityTest(BaseCompatibilityTestCase):
 
-    image_model = 'open_clip/ViT-B-32/laion2b_s34b_b79k'
+    image_model = 'Marqo/ViT-B-32/laion2b_s34b_b79k'
```
I think we should test both not stop testing the old one. Alternatively, I think we can have a unit test with model pair names, and simply assert that other than the name they're exactly identical in the registry.

farshidz replied:
The above works for OpenCLIP models. HF models have a change of type so need to be tested separately

adityabharadwaj198 replied:
Added both types of tests

farshidz commented on tests/s2_inference/embeddings_reference/embeddings_open_clip_python_3_8.json:
```diff
@@ -1 +1 @@
-{"open_clip/RN50/yfcc15m": {"hello": [[-0.31743431091308594, 0.3105415403842926, -0.15373371541500092, 0.27386870980262756, 0.4532639980316162, -0.20620116591453552, 0.32274773716926575, -0.3638650178909302, -0.19914303719997406, 0.47483378648757935, 0.6378846764564514, -0.29982784390449524, -0.7905068397521973, -0.09241227060556412, -0.10728361457586288, 0.1741500198841095, -0.07469315081834793, ...
\ No newline at end of file
+{"open_clip/RN50/yfcc15m": {"hello": [[-0.31743431091308594, 0.3105415403842926, -0.15373371541500092, 0.27386870980262756, 0.4532639980316162, -0.20620116591453552, 0.32274773716926575, -0.3638650178909302, -0.19914303719997406, 0.47483378648757935, 0.6378846764564514, -0.29982784390449524, -0.7905068397521973, -0.09241227060556412, -0.10728361457586288, 0.1741500198841095, -0.07469315081834793, ...
```
Need your help to understand what's going on here

adityabharadwaj198 replied:
there are new models added so I've added embeddings corresponding to those

wanliAlex commented on src/marqo/s2_inference/model_registry.py:
```diff
@@ -144,26 +144,51 @@ def _get_open_clip_properties() -> Dict:
                                              'note': 'open_clip models',
                                              'type': 'open_clip',
                                              'pretrained': 'laion400m_e31'},
+        'Marqo/ViT-B-32/laion400m_e31': {'name': 'open_clip/ViT-B-32/laion400m_e31',
```
I am checking the [marqtune docs](https://docs.marqo.ai/latest/models/marqtune/list-of-models/). Should the model name be `Marqo/ViT-B-32.laion400m_e31`?

adityabharadwaj198 replied:
changed acc to marqtune

wanliAlex commented on src/marqo/s2_inference/model_registry.py:
```diff
@@ -214,21 +259,41 @@ def _get_open_clip_properties() -> Dict:
                                              'note': 'open_clip models',
                                              'type': 'open_clip',
                                              'pretrained': 'laion400m_e31'},
+        'Marqo/ViT-L-14/laion400m_e31': {'name': 'open_clip/ViT-L-14/laion400m_e31',
```
Use `{"name": "hf-hub:timm/vit_large_patch14_clip_224.laion400m_e31"}` to load the model

wanliAlex replied:
https://github.com/mlfoundations/open_clip/blob/aeaf2a063b0b6df2140bd11617b717da6bedab67/src/open_clip/pretrained.py#L271

adityabharadwaj198 replied:
done

wanliAlex commented on src/marqo/s2_inference/model_registry.py:
```diff
@@ -817,6 +953,14 @@ def _get_hf_properties() -> Dict:
+                 "dimensions": 768,
+                 "tokens": 512,
+                 "type": "sbert",
+                 "poolingMethod": "mean",
```
This should be `cls`. Check [here](https://huggingface.co/BAAI/bge-base-en-v1.5/blob/main/1_Pooling/config.json#L3). This comment also applied to other models.

adityabharadwaj198 replied:
discussed offline

wanliAlex commented on src/marqo/s2_inference/model_registry.py:
```diff
@@ -817,6 +953,14 @@ def _get_hf_properties() -> Dict:
+                {"name": 'BAAI/bge-base-en-v1.5',
+                 "dimensions": 768,
+                 "tokens": 512,
+                 "type": "sbert",
```
They type should be `"hf"`

adityabharadwaj198 replied:
type was sbert. Moved them under sbert method

adityabharadwaj198 committed:
Addressing comments # 1

adityabharadwaj198 committed:
added embeddings for the new sbert models but they don't work

adityabharadwaj198 committed:
fixing tests

adityabharadwaj198 committed:
Changing one model's name which was having issues.

Known error: https://github.com/mlfoundations/open_clip/issues/594

Error: https://github.com/marqo-ai/marqo/actions/runs/12275146990/job/34249731062#step:9:1245

farshidz commented on tests/s2_inference/embeddings_reference/embeddings_open_clip_python_3_8.json:
```diff
@@ -1 +1 @@
-{"open_clip/RN50/yfcc15m": {"hello": [[-0.31743431091308594, 0.3105415403842926, -0.15373371541500092, 0.27386870980262756, 0.4532639980316162, -0.20620116591453552, 0.32274773716926575, -0.3638650178909302, -0.19914303719997406, 0.47483378648757935, 0.6378846764564514, -0.29982784390449524, -0.7905068397521973, -0.09241227060556412, -0.10728361457586288, 0.1741500198841095, -0.07469315081834793, ...
\ No newline at end of file
+{"open_clip/RN50/yfcc15m": {"hello": [[-0.31743431091308594, 0.3105415403842926, -0.15373371541500092, 0.27386870980262756, 0.4532639980316162, -0.20620116591453552, 0.32274773716926575, -0.3638650178909302, -0.19914303719997406, 0.47483378648757935, 0.6378846764564514, -0.29982784390449524, -0.7905068397521973, -0.09241227060556412, -0.10728361457586288, 0.1741500198841095, -0.07469315081834793, ...
```
nit. we add lines to those json files to the diff makes more sense in the future

adityabharadwaj198 replied:
i did just that, idk why it shows a completely new file. maybe something to do with a huge number of additions?

farshidz commented on tests/s2_inference/test_model_registry.py:
```diff
@@ -0,0 +1,66 @@
+        self.open_clip_properties = _get_open_clip_properties()
+
+    def test_open_clip_model_configurations_match(self):
+        with self.subTest("Check if Marqo/ViT-B-32.openai & open_clip/ViT-B-32/openai configurations match"):
```
This is hard to read and maintain. I suggest having a list of pairs to test first, then one loop to test them all. e.g.

models_pairs = [('Marqo/ViT-B-32.openai', 'open_clip/ViT-B-32/openai'), ...]

for pair in model_pairs:
    with self.subTest(...

Furthermore, please add a description to this test otherwise in a few months' time we will have no idea why we have such a test.

adityabharadwaj198 replied:
good idea, incorporated

adityabharadwaj198 committed:
addressing comments # 2

adityabharadwaj198 committed:
Fixing keyError
